### ISSUE TITLE
2.9.23 Breaks then-redis

### ISSUE THREAD

**Issue by JSteunou** [petkaantonov/bluebird#561]
and maybe others lib. I got this error

```
Possibly unhandled TypeError: Cannot redefine property: usingDomains
    at Function.defineProperty (native)
    at Object.<anonymous> (/home/jerome/someproject/node_modules/bluebird/js/main/async.js:75:16)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
    at module.exports (/home/jerome/someproject/node_modules/bluebird/js/main/promise.js:13:13)
    at Object.<anonymous> (/home/jerome/someproject/node_modules/bluebird/js/main/bluebird.js:9:39)
    at Module._compile (module.js:456:26)
    at Object.Module._extensions..js (module.js:474:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.require (module.js:364:17)
    at require (module.js:380:17)
```

**Comment by phpnode** [comment]
dupe of #559

**Comment by petkaantonov** [comment]
Fixed in 2.9.24
